src/main/java/build/buildfarm/common/redis/RedisNodeHashes.java

```diff
@@ -104,7 +111,15 @@ public static List<String> getEvenlyDistributedHashesWithPrefix(
   private static Iterable<List<List<Long>>> getNodeSlotRanges(UnifiedJedis jedis) {
     if (jedis instanceof JedisCluster) {
       // get slot range information for each shard
-      return transform(getClusterShards((JedisCluster) jedis), ClusterShardInfo::getSlots);
+      try {
+        return transform(getClusterShards((JedisCluster) jedis), ClusterShardInfo::getSlots);
+      } catch (RuntimeException re) {
+        log.log(
+            Level.WARNING,
+            "Unable to get Redis Cluster shards. Fall-back to deprecated implementation",
```

Comment:
nit: "CLUSTER SHARDS failed. Falling back to CLUSTER NODES (deprecated)"

There's enough context in the backtrace to implicate redis.